packages/core/js-client/src/jsServiceHost/JsServiceHost.ts

```diff
@@ -13,99 +13,125 @@
+
+  /**
+   * Returns true if any handler for the specified serviceId is registered
+   */
+  hasService(serviceId: string): boolean {
+    return (
+      this.commonHandlers.has(serviceId) ||
+      this.particleScopeHandlers.has(serviceId)
+    );
+  }
+
+  /**
+   * Removes all handlers associated with the specified particle scope
+   * @param particleId Particle ID to remove handlers for
+   */
+  removeParticleScopeHandlers(particleId: string): void {
+    this.particleScopeHandlers.delete(particleId);
+  }
+
+  /**
+   * Find call service handler for specified particle
+   * @param serviceId Service ID as specified in `call` air instruction
+   * @param fnName Function name as specified in `call` air instruction
+   * @param particleId Particle ID
+   */
+  getHandler(
+    serviceId: string,
+    fnName: string,
+    particleId: string,
+  ): GenericCallServiceHandler | null {
+    const key = serviceFnKey(serviceId, fnName);
+    const psh = this.particleScopeHandlers.get(particleId);
+    let handler: GenericCallServiceHandler | undefined = undefined;
+
+    // we should prioritize handler for this particle if there is one
+    // if particle-scoped handler exist for this particle try getting handler there
+    if (psh !== undefined) {
+      handler = psh.get(key);
     }
 
-    /**
-     * Removes all handlers associated with the specified particle scope
-     * @param particleId Particle ID to remove handlers for
-     */
-    removeParticleScopeHandlers(particleId: string): void {
-        this.particleScopeHandlers.delete(particleId);
+    // then try to find a common handler for all particles with this service-fn key
+    // if there is no particle-specific handler, get one from common map
+    if (handler === undefined) {
+      handler = this.commonHandlers.get(key);
     }
 
-    /**
-     * Find call service handler for specified particle
-     * @param serviceId Service ID as specified in `call` air instruction
-     * @param fnName Function name as specified in `call` air instruction
-     * @param particleId Particle ID
-     */
-    getHandler(serviceId: string, fnName: string, particleId: string): GenericCallServiceHandler | null {
-        const key = serviceFnKey(serviceId, fnName);
-        const psh = this.particleScopeHandlers.get(particleId);
-        let handler: GenericCallServiceHandler | undefined = undefined;
-
-        // we should prioritize handler for this particle if there is one
-        // if particle-scoped handler exist for this particle try getting handler there
-        if (psh !== undefined) {
-            handler = psh.get(key);
-        }
-
-        // then try to find a common handler for all particles with this service-fn key
-        // if there is no particle-specific handler, get one from common map
-        if (handler === undefined) {
-            handler = this.commonHandlers.get(key);
-        }
-
-        return handler || null;
+    return handler ?? null;
```

Comment:
```suggestion
    return psh?.get(key) ?? this.commonHandlers.get(key) ?? null;
```

Reply:
Nice! This suggestion is invalid but i will apply this manually